### ISSUE TITLE
BUG: np.select cannot does not return values in choicelist when default used is empty container

### ISSUE THREAD

**Issue by gitgithan** [numpy/numpy#21092]
### Describe the issue:

I want to have empty list as default value of np.select.

When the default value is selected to be an empty list or tuple, `choicelist` will be overwritten from `[array(1), array(2), array(3), array([], dtype=float64)]`  to `[array([], dtype=int64), array([], dtype=int64), array([], dtype=int64), array([], dtype=float64)]` 
after `choicelist = np.broadcast_arrays(*choicelist)` in `function_base.py`.

The result is that I get `array([], dtype=float64)` returned when i expect 3 to be returned (because the conditions are False,False,True corresponding to 1,2,3).

### Reproduce the code example:

```python
import numpy as np

conds = [False, 
         False, 
         True]

print(np.select(conds, [1,2,3], default=[]))
```
```


### Error message:

_No response_

### NumPy/Python version information:

1.21.4 3.8.12 (default, Dec  8 2021, 11:22:00) 
[Clang 12.0.5 (clang-1205.0.22.11)]

**Comment by soma2000-lang** [comment]
@gitgithan  working on this

**Comment by Adarshsng** [comment]
Hello @gitgithan,

It's definitely happening due to broadcasting, which is there to provide the shape i.e. needed for the results.
Over that, if you look at the doc

https://numpy.org/doc/stable/reference/generated/numpy.select.html

it clearly says that the default value needs to be a scalar quantity, therefore feeding in a list that is empty or has elements of more than 1, causes issues with the broadcasting.

Do let me know if the community is thinking of enhancing codes so as to apply solutions to such edge cases.

**Comment by seberg** [comment]
Yeah, but the code seems to generally allow non-scalars and broadcast all choices (including the default!).
Broadcasting can be surprising with empty arrays (arrays with a dimension of size 0), since it "removes" elements.  In principle, we could possibly avoid it, but I wonder if may not also make sense to adjust the documentation and just say that the default is broadcast together with the condlist.
